src/Raven.Server/Documents/Sharding/ShardedDatabaseContext.Streaming.cs

```diff
@@ -210,7 +210,23 @@ public override int Compare(ShardStreamItem<BlittableJsonReaderObject> x, ShardS
 
                 public static DocumentLastModifiedComparer Instance = new ();
             }
-            
+
+            public sealed class RevisionLastModifiedComparer : Comparer<ShardStreamItem<BlittableJsonReaderObject>>
+            {
+                public override int Compare(ShardStreamItem<BlittableJsonReaderObject> x, ShardStreamItem<BlittableJsonReaderObject> y)
+                {
+                    if (x.Item.TryGet(nameof(Document.LastModified), out DateTime xLastModified) == false)
+                        throw new InvalidOperationException($"Revision does not contain 'LastModified' field.");
+
+                    if (x.Item.TryGet(nameof(Document.LastModified), out DateTime yLastModified) == false)
```

Comment:
this should be `y.Item`, no? don't duplicate code, make a method out of this.

Reply:
Done